### PR TITLE
specification for cluster list view

### PR DESCRIPTION
**Pull request by gnehapk**
@anivargi @brainfunked @nthomas-redhat @kamleshverma1 @anmolbabu Please review.

### CODE REVIEW

File: specs/cluster_list_view.adoc [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+==== Tendrl API impact:
+
+/api/1.0/GetClusterList
+
+Required Input from Anup - 
+
+* To add the response format for the API call
+* To add the polling API call and its response format	
+
```
Comment (gnehapk) [inline specs/cluster_list_view.adoc:67]: @anivargi Please provide the required inputs for API calls.

**Comment by gnehapk** [comment]
Addresses #75.

File: specs/cluster_list_view.adoc [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+== Problem description
+
+Tendrl should show list of Ceph and Gluster clusters present by Tendrl along with their details.
+
+The details shown for each cluster in UI is -
+
+* Cluster Name
+* Status
+* Storage Usage
+* CPU Utilisation
```
Comment (cybernth) [inline specs/cluster_list_view.adoc]: typo? is it CPU utilization?

File: specs/cluster_list_view.adoc [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+Users can't perfrom filter and sort operations because name, provider, status and utilisation data is not supported by API. 
+
+== Proposed change
+
+* Create a view for listing cluster details.
+* For showing storage usage and cpu utilisation, use donut chart component.
+* Implement client side filer and sorting functionality.
+* Filter call will trigger after 1 sec the user enters the filter data.
+* Send polling call after 10 mins to get the updated data.
```
Comment (cybernth) [inline specs/cluster_list_view.adoc]: 10 mins?

File: specs/cluster_list_view.adoc [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+Generate list component can be created and used for listing the cluster. But because of its complexity and time constraint, we will pick this up later.
+
+=== Data model impact:
+
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+/api/1.0/GetClusterList
```
Comment (cybernth) [inline specs/cluster_list_view.adoc]: This part lack information. state what is available currently(cluster wide list API available), what extra information reqd in the current API. Also specify the any extra APIs required(syatem wide listing?)
Which component implements sorting and filtering?
Any APis required to get the utlization data?

File: specs/cluster_list_view.adoc [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+Test whether clusters are correctly being displayed or not.
+
+== Documentation impact:
+
+None
+
+== References:
+
+https://redhat.invisionapp.com/share/BR8JDCGSQ#/screens/185937524
```
Comment (cybernth) [inline specs/cluster_list_view.adoc:139]: Add a reference to topic issue as well here - https://github.com/Tendrl/specifications/issues/75

File: specs/cluster_list_view.adoc [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+/api/1.0/GetClusterList
+
+Required Input from Anup - 
+
+* To add the response format for the API call
+* To add the polling API call and its response format	
+
+==== Notifications/Monitoring impact:
+
+Monitoring data support is not present for cluster list.
```
Comment (cybernth) [inline specs/cluster_list_view.adoc:70]: what about cluster utilization?

**Review by cybernth — CHANGES_REQUESTED** [review]

**Comment by gnehapk** [comment]
@cloudbehl @anivargi I have updated the PR. Please review it.

File: specs/cluster_list_view.adoc [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+== Problem description
+
+Tendrl should show list of Ceph and Gluster clusters present by Tendrl along with their details.
+
+The details shown for each cluster in UI is -
+
+* Cluster Name
+* Status
+* Storage Usage
+* Utilisation
```
Comment (ltrilety) [inline specs/cluster_list_view.adoc]: Is it CPU utilization?
Anyway is it a good information at all to have CPU or memory utilization for cluster? What does it say about cluster?

**Review by cloudbehl — APPROVED** [review]

**Comment by gnehapk** [comment]
@anivargi For the cluster list view, below are the data that is not present in the json response of cluster -
1) Status
2) IOPS
3) Host count
4) Pool count
5) Last Sync
6) Provider

These are the data(with their mapped attribute) which are provided by API - 
1) name - maps.mon_status.cluster_name
2) Storage(donut chart) - stats.storage
3) Alert count - stats.alert_cnt

Please confirm, whether these mappings are correct or not?

**Comment by mkudlej** [comment]
Please add https://github.com/Tendrl/usmqe-tests/issues/16 for testing reference.

**Pull request merged**